### PR TITLE
Sigma zdc

### PR DESCRIPTION
**Pull request by sebouh137**
### Briefly, what does this PR introduce?
Creates a benchmark for reconstructing Σ0 baryons in the ZDC.  

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [X] Other: new benchmark

### Please check if this PR fulfills the following:
- [X] Tests for the changes have been added
- [X] Documentation has been added / updated
- [X] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
no
### Does this PR change default behavior?
yes.  Adds a new benchmark to the pantheon of benchmarks

### CODE REVIEW

File: benchmarks/sigma/config.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    matrix:
+      - P: 100
+      - P: 125
+      - P: 150
+      - P: 175
+      - P: 200
+      - P: 225
+      - P: 250
+      - P: 275
+  timeout: 6 hours
```
Comment (wdconinc) [inline benchmarks/sigma/config.yml:16]:
```suggestion
  timeout: 1 hours
```

**Comment by sebouh137** [comment]
@wdconinc Let me know if there's anything else you'd like me to include or change in this benchmark.  All the checks (except for the DIS one, which is outside the scope of this pull request) have passed.  Thanks, Sebouh

**Comment by wdconinc** [comment]
It doesn't actually get run: https://eicweb.phy.anl.gov/EIC/benchmarks/physics_benchmarks/-/pipelines/100489

**Comment by sebouh137** [comment]
> It doesn't actually get run: https://eicweb.phy.anl.gov/EIC/benchmarks/physics_benchmarks/-/pipelines/100489 (ignore the dis failure which is understood and the topic of #22).

wowzers!   Wouter, you're right!  Ok, I think the recent change to .gitlab-ci.yml will fix this.

**Comment by wdconinc** [comment]
I'm going to let @veprbl give the last word here. I just want it to run fast enough so we don't have to wait 2 hours for stuff to merge. But hopefully this is all due to the hcal zdc clustering.

**Comment by veprbl** [comment]
This is what I get when running locally
```
Traceback (most recent call last):
  File "benchmarks/sigma/analysis/sigma_plots.py", line 81, in <module>
    is_neutron_cand[p][i][index_of_max]=1
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
IndexError: list assignment index out of range
```

**Comment by sebouh137** [comment]
> This is what I get when running locally
> 
> ```
> Traceback (most recent call last):
>   File "benchmarks/sigma/analysis/sigma_plots.py", line 81, in <module>
>     is_neutron_cand[p][i][index_of_max]=1
>     ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
> IndexError: list assignment index out of range
> ```

huh... I think what's going on is that if there's an event with zero clusters and it is trying to assign the one with the maximum eigenvalue as the neutron, it is unable to do so.  I'll add an if statement to skip this if there are no clusters in the event.

**Comment by sebouh137** [comment]
> > This is what I get when running locally
> > ```
> > Traceback (most recent call last):
> >   File "benchmarks/sigma/analysis/sigma_plots.py", line 81, in <module>
> >     is_neutron_cand[p][i][index_of_max]=1
> >     ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
> > IndexError: list assignment index out of range
> > ```
> 
> huh... I think what's going on is that if there's an event with zero clusters and it is trying to assign the one with the maximum eigenvalue as the neutron, it is unable to do so. I'll add an if statement to skip this if there are no clusters in the event.

@veprbl Actually, this hypothesis is incorrect.  I think what is actually happening is that the number of shape parameters per cluster is not the same as the value I had hardcoded, which was 7.  So, I decided to fix this by having the code calculate the number of shape parameters per cluster by dividing the total number of shape parameters by the number of clusters.  This way if anyone adds any new shape parameters or removes any of the ones I am not using, it won't mess up my code.  Please rerun the sigma_plots code and see if this change fixes your results.  Thanks

**Comment by veprbl** [comment]
I'm curious, so you call a "neutron" whatever has largest "width"? What is the idea behind that? Is it to select the longest cluster?

**Comment by sebouh137** [comment]
> I'm curious, so you call a "neutron" whatever has largest "width"? What is the idea behind that? Is it to select the longest cluster?

That is correct.   My expected signal in this analysis is 1 neutron and 3 photons.  Therefore, I am selecting the cluster that has the longest extent in any direction as the neutron, since the neutron shower is expected to have a much larger longitudinal extent than the photon showers.

**Comment by sebouh137** [comment]
@veprbl Have you had any success with running the benchmark locally with the updated version of the sigma_plots.py?

**Comment by veprbl** [comment]
@sebouh137 It works now. We still need the fix in EICrecon to fix the reconstruction performance.

**Comment by sebouh137** [comment]
@wdconinc @veprbl Why is the sigma:analyze stage of the check rerunning everything from the sigma:simulate?  Shouldn't the files created in the sigma:simulate stage propagate to the sigma:analyze?

**Comment by sebouh137** [comment]
> @wdconinc @veprbl Why is the sigma:analyze stage of the check rerunning everything from the sigma:simulate? Shouldn't the files created in the sigma:simulate stage propagate to the sigma:analyze?

Update, apparently Snakemake thinks that the simulation needs to be redone because the gen_sigma_dec.cxx has been "updated".   This is what I get from the logs (see below).  Why is it doing this?  

[32;1m$ snakemake --cores 1 results/epic_zdc_sipm_on_tile_only/sigma[0;m
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Job stats:
job               count    min threads    max threads
--------------  -------  -------------  -------------
sigma_analysis        1              1              1
sigma_generate        8              1              1
sigma_recon           8              1              1
sigma_simulate        8              1              1
total                25              1              1

Select jobs to execute...

[Sat Aug 24 00:08:56 2024]
rule sigma_generate:
    input: benchmarks/sigma/analysis/gen_sigma_decay.cxx
    output: sim_output/sigma/sigma_decay_175GeV.hepmc
    jobid: 12
    reason: Updated input files: benchmarks/sigma/analysis/gen_sigma_decay.cxx
    wildcards: P=175
    resources: tmpdir=/tmp

**Pull request merged**